Extensions/Optimizely/Src/Tasks/OptimizelyX/operations/TaskOperation.ts

```diff
@@ -0,0 +1,196 @@
+import { OptimizelyXClient, IOptimizelyAlphaBetaTest, IOptimizelyEnvironment, OptimizelyExperimentStatus } from './OptimizelyXClient';
+import { OptimizelyTaskParameter } from './../models/OptimizelyTaskParameter'
+import * as tl from 'azure-pipelines-task-lib/task';
+
+export class TaskOperation {
+
+  private optimizelyClient: OptimizelyXClient;
+  private taskInputs: OptimizelyTaskParameter;
+
+  constructor(taskInputs: OptimizelyTaskParameter) {
+    this.taskInputs = taskInputs;
+    this.optimizelyClient = this.createOptimizelyClientInstance();
+  }
+
+  public async startABTest(experiment: IOptimizelyAlphaBetaTest): Promise<void> {
+      let environmentName: string = this.taskInputs.getEnvironmentName();
+      let projectId: string = this.taskInputs.getProjectId();
+
+      if(this.taskInputs.isWebProject()) {
+          experiment.status = OptimizelyExperimentStatus.Running;
+      } else {
+          let environment: IOptimizelyEnvironment = await this.optimizelyClient.getEnvironment(projectId, environmentName);
+          if (environment == null) {
+              throw tl.loc("EnvironmentNotFound", environmentName);
+          }
+
+          Object.keys(experiment.environments).forEach((value) => {
+              if (value.toLowerCase() == environmentName.toLowerCase()) {
+                  experiment.environments[value].status = OptimizelyExperimentStatus.Running;
+              }
+          });
+      }
+
+      let experimentName = experiment.name;
+      await this.optimizelyClient.updateExperiment(experiment.id, experiment);
+
+      let counter: number = 6;
+      while (counter > 0) {
+          console.log(tl.loc("WaitingForExperimentToStart", experimentName));
+          await this.sleep(10000);
+          counter--;
+      }
```

Comment:
Can't we poll instead of this iterative sleep?

Reply:
REST API update configuration of the experiment and data file corresponding to experiment got updated after approx 1 min. So, we can't check status using REST API. Though we can get data file corresponding to the project and identify if experiment has been started or paused. Little complex scenario here will be if experiment state was as previous but only variation has been changed.